### PR TITLE
New serverless pattern - AWS Secrets Manager to CloudWatch Events to SNS (Terraform)

### PR DESCRIPTION
**Pull request by MakendranG**
*Issue #, if available:* https://github.com/aws-samples/serverless-patterns/issues/1723

*Description of changes:*

This pattern contains a terraform template to detect and notify on Amazon Secrets Manager Secret Key Creation, Updation and Deletion using Amazon CloudWatch event and Amazon SNS.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by MakendranG — COMMENTED** [review]
Updated the pattern file.

**Review by MakendranG — COMMENTED** [review]
Updated the file

**Comment by MakendranG** [comment]
The example-pattern file have all the parameters. Still failing the checks. Kindly update which parameter is getting failed.

File: sm-cw-lambda-sns-terraform/example-pattern.json [hunk 1]
```diff
@@ -0,0 +1,57 @@
+{
+  "title": "AWS Secrets Manager to CloudWatch Events to Lambda to SNS",
+  "description": "An terraform template that detects and notifies on Amazon Secrets Manager secret key creation, Updation and deletion using Amazon CloudWatch event and Amazon SNS",
+  "language": "YAML",
+  "level": "200",
+  "framework": "Terraform",
+  "introBox": {
+    "headline": "How it works",
+    "text": [
+      "Once the template deployed successfully, first thing to do is to confirm the Email subscription. You will receive an email to confirm it.  Then, head to AWS Secrets Manager console and Create a Secret Key by following these steps - https://docs.aws.amazon.com/secretsmanager/latest/userguide/create_secret.html . That's it. You will soon receive a notification about the key you created. Now any event (Like Update or Delete) happens related to that Secret Key, you will receive the notification.",
```
Comment (mavi888) [inline sm-cw-lambda-sns-terraform/example-pattern.json:10]: i think the comma at the end of this line is the one causing problems, remove it and we can run the validation again.

**Review by MakendranG — COMMENTED** [review]
removed the dot in the pattern file

**Comment by MakendranG** [comment]
Hello @mavi888 

Thanks for the update.
As per the request, I have updated the pattern file. Automated system have picked the pattern and validated. Now it is waiting for approval. Kindly do the needful.

**Comment by mavi888** [comment]
I just tested this pattern and it doesnt work. 
When looking at the rule created there is no target defined for it.

File: sm-cw-lambda-sns-terraform/README.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    ```
+1. During the prompts:
+    * Provide your email address to receive notification from SNS:
+    * Enter yes
+1. Confirm all created resources has been deleted
+    ```bash
+    terraform show
+    ```
+----
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (mavi888) [inline sm-cw-lambda-sns-terraform/README.md:70]: ----
Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: MIT-0

This is the way to finalize the read me file with the copyright.

File: sm-cw-lambda-sns-terraform/example-pattern.json (hunk 1 above)
Comment (MakendranG) [reply sm-cw-lambda-sns-terraform/example-pattern.json]: I have removed it and validation successfully passed

**Comment by MakendranG** [comment]
> I just tested this pattern and it doesnt work. When looking at the rule created there is no target defined for it.

Have modified the terraform file with the target.

File: sm-cw-lambda-sns-terraform/README.md (hunk 2 above)
Comment (MakendranG) [reply sm-cw-lambda-sns-terraform/README.md:71]: Have added the readme file as requested

**Comment by MakendranG** [comment]
Hello @mavi888

Thanks for the update.
As per the request, I have updated the terraform file. Automated system have picked the pattern and validated. Now it is waiting for approval. Kindly do the needful.

File: sm-cw-lambda-sns-terraform/README.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+   -   Provide your email address to receive notification from SNS:
+   -   Enter yes
+## How it works
+
+This template is used to monitor AWS Secrets Manager secret keys. This helps in reporting when something is wrong, and take automatic actions when appropriate. Once the template is deployed, you will receive an email notification on the email address you defined. Make sure to confirm email subscription in order to receive updates related to your Secret Keys present in AWS Secret manager.
+
+## Testing
+
+Once the template deployed successfully, first thing to do is to confirm the Email subscription. You will receive an email to confirm it.  Then, head to AWS Secrets Manager console and Create a Secret Key by following these steps - https://docs.aws.amazon.com/secretsmanager/latest/userguide/create_secret.html . That's it. You will soon receive a notification about the key you created. Now any event (Like Update or Delete) happens related to that Secret Key, you will receive the notification.
+Log into the AWS Console, browse to AWS IoT Core:
```
Comment (mavi888) [inline sm-cw-lambda-sns-terraform/README.md:49]: what is this last line about?

**Comment by mavi888** [comment]
The pattern doesnt work.
Please test it before submiting it again

**Review by MakendranG — COMMENTED** [review]
Made the necessary changes

**Review by MakendranG — COMMENTED** [review]
Modified pattern file

**Comment by mavi888** [comment]
This pattern still doesnt work. I will close it and please open it when you are sure you test it out, granted all the permissions needed and everything is polished.

**Pull request closed without merging**